src/index.js

```diff
@@ -289,11 +291,12 @@ export default function undoable (reducer, rawConfig = {}) {
           }
         }
 
-        if (history.present !== res) {
+        // Don't add entry to history if it was just created
+        if ((justCreatedHistory && history.present === res) || action.type === '@@redux-undo/INIT') {
```

Comment:
Well that is the question.. you think you could make HMR with both initial state from stores and reducers work without touching `@@redux/INIT`? I'm not sure if I know how.